src/main.rs

```diff
@@ -337,58 +355,46 @@ fn process_file(
-        if line.trim() == "#[rustfmt::skip]" {
+        if trimmed == "#[rustfmt::skip]" {
+            println!("Skipping rustfmt");
             continue;
         }
 
-        if line.trim().starts_with("#REPLACE ") {
-            let what = line.trim().strip_prefix("#REPLACE ").unwrap();
+        // Check if we should replace the next line with the key/value of a variable
+        if let Some(what) = trimmed
+            .strip_prefix("#REPLACE ")
+            .or_else(|| trimmed.strip_prefix("//REPLACE "))
+        {
             let mut split = what.split_terminator(" ");
             replace = Some(split.next().unwrap().to_string());
             let var = split.next().unwrap().to_string();
-            for (key, value) in variables {
-                if key == &var {
-                    replacement = Some(value);
-                    break;
-                }
-            }
-        } else if line.trim().starts_with("//REPLACE ") {
-            let what = line.trim().strip_prefix("//REPLACE ").unwrap();
-            let mut split = what.split_terminator(" ");
-            replace = Some(split.next().unwrap().to_string());
-            let var = split.next().unwrap().to_string();
-            for (key, value) in variables {
-                if key == &var {
-                    replacement = Some(value);
-                    break;
-                }
-            }
-        } else if line.trim().starts_with("#IF ") {
-            let cond = line.trim().strip_prefix("#IF ").unwrap();
-            if let Some(stripped) = cond.strip_prefix("!") {
-                include.push(!options.contains(&stripped.to_string()) && *include.last().unwrap());
-            } else {
-                include.push(options.contains(&cond.to_string()) && *include.last().unwrap());
+
+            // Find the replacement value from the variables map
+            if let Some((_, value)) = variables.iter().find(|(key, _)| key == &var) {
+                replacement = Some(value);
             }
-        } else if line.trim().starts_with("#ENDIF") {
-            include.pop();
-        } else if line.trim().starts_with("//IF ") {
-            let cond = line.trim().strip_prefix("//IF ").unwrap();
-            if let Some(stripped) = cond.strip_prefix("!") {
-                include.push(!options.contains(&stripped.to_string()) && *include.last().unwrap());
+        // Check if we should include the next line(s)
+        } else if trimmed.starts_with("#IF ") || trimmed.starts_with("//IF ") {
+            let cond = if trimmed.starts_with("#IF ") {
+                trimmed.strip_prefix("#IF ").unwrap()
             } else {
-                include.push(options.contains(&cond.to_string()) && *include.last().unwrap());
-            }
-        } else if line.trim().starts_with("//ENDIF") {
+                trimmed.strip_prefix("//IF ").unwrap()
+            };
+
+            let cond = process_contains_option(cond.to_string(), &mut scope, &engine);
```

Comment:
we won't need that with the change of the registered function

Reply:
we now can remove this `process_contains_option` and also remove the function

Reply:
Awesome! I definitely overcomplicated things :sweat:

Reply:
I probably did a bad job explaining the idea about Rhai ... anyways looks good now!